tests/foreman/ui/test_host.py

```diff
@@ -626,8 +632,31 @@ def test_negative_delete_primary_interface(session, module_host_template):
     :CaseLevel: System
     """
     interface_id = gen_string('alpha')
+    host_template = entities.Host(organization=module_org, location=module_location)
+    host_template.create_missing()
+    interface = (
+        'type=interface,mac={},identifier=eth0,name={},domain_id={},'
+        'ip={},primary=true,provision=true'
+    ).format(host_template.mac, gen_string('alpha'), host_template.domain.id, gen_ipaddr())
+    os_name = f'{host_template.operatingsystem.name} {host_template.operatingsystem.major}'
     with session:
-        host_name = create_fake_host(session, module_host_template, interface_id=interface_id)
+        host = make_fake_host(
```

Comment:
Given the test coverage is concerned with the delete action, the host create does not have to happen through the UI so this is okay.

It should happen in a fixture though, not in the test function itself - because its test setup, and isn't being done via the UI.